docs/sql-ref-datetime-pattern.md

```diff
@@ -26,44 +26,128 @@ There are several common scenarios for datetime usage in Spark:
 - Datetime functions related to convert `StringType` to/from `DateType` or `TimestampType`.
   For example, `unix_timestamp`, `date_format`, `to_unix_timestamp`, `from_unixtime`, `to_date`, `to_timestamp`, `from_utc_timestamp`, `to_utc_timestamp`, etc.
 
-Spark uses pattern letters in the following table for date and timestamp parsing and formatting:
-
-|Symbol|Meaning|Presentation|Examples|
-|------|-------|------------|--------|
-|**G**|era|text|AD; Anno Domini; A|
-|**y**|year|year|2020; 20|
-|**D**|day-of-year|number|189|
-|**M/L**|month-of-year|number/text|7; 07; Jul; July; J|
-|**d**|day-of-month|number|28|
-|**Q/q**|quarter-of-year|number/text|3; 03; Q3; 3rd quarter|
-|**Y**|week-based-year|year|1996; 96|
-|**w**|week-of-week-based-year|number|27|
-|**W**|week-of-month|number|4|
-|**E**|day-of-week|text|Tue; Tuesday; T|
-|**u**|localized day-of-week|number/text|2; 02; Tue; Tuesday; T|
-|**F**|week-of-month|number|3|
-|**a**|am-pm-of-day|text|PM|
-|**h**|clock-hour-of-am-pm (1-12)|number|12|
-|**K**|hour-of-am-pm (0-11)|number|0|
-|**k**|clock-hour-of-day (1-24)|number|0|
-|**H**|hour-of-day (0-23)|number|0|
-|**m**|minute-of-hour|number|30|
-|**s**|second-of-minute|number|55|
-|**S**|fraction-of-second|fraction|978|
-|**V**|time-zone ID|zone-id|America/Los_Angeles; Z; -08:30|
-|**z**|time-zone name|zone-name|Pacific Standard Time; PST|
-|**O**|localized zone-offset|offset-O|GMT+8; GMT+08:00; UTC-08:00;|
-|**X**|zone-offset 'Z' for zero|offset-X|Z; -08; -0830; -08:30; -083015; -08:30:15;|
-|**x**|zone-offset|offset-x|+0000; -08; -0830; -08:30; -083015; -08:30:15;|
-|**Z**|zone-offset|offset-Z|+0000; -0800; -08:00;|
-|**'**|escape for text|delimiter| |
-|**''**|single quote|literal|'|
-|**[**|optional section start| | |
-|**]**|optional section end| | |
-
-The count of pattern letters determines the format.
-
-- Text: The text style is determined based on the number of pattern letters used. Less than 4 pattern letters will use the short form. Exactly 4 pattern letters will use the full form. Exactly 5 pattern letters will use the narrow form. Six or more letters will fail.
+The following tables define how the pattern letters be used for date and timestamp parsing and formatting in Spark.
+
+## Date Fields
+
+Pattern letters to output a date:
+
+|Pattern|Count|Meaning|Presentation|Examples|
+|---|---|---|---|---|
+|**G**|1|era|text|AD|
+|**GG**|2|era|text|AD|
+|**GGG**|3|era|text|AD|
```

Comment:
can be simplified as `G..G | 1-3 `?

Reply:
I tried this once, but some of the patterns are not able to be simplified because of different outputs. So I choose to stick to same layout.